fee/src/lib.rs

```diff
@@ -1,19 +1,43 @@
-use {solana_fee_structure::FeeDetails, solana_svm_transaction::svm_message::SVMMessage};
+use {
+    solana_feature_set::{
+        enable_secp256r1_precompile, remove_rounding_in_fee_calculation, FeatureSet,
+    },
+    solana_fee_structure::FeeDetails,
+    solana_svm_transaction::svm_message::SVMMessage,
+};
+
+/// Bools indicating the activation of features relevant
+/// to the fee calculation.
+#[derive(Copy, Clone)]
+pub struct FeeFeatures {
```

Comment:
Goal of this structure is so that we do not have a bunch of bool arguments that we always need, but also to not introduce the entire feature set which.
Keeping this makes it extremely obvious exactly which features affect the fee-calculation; and the `FeeFeatures` could even be re-used to avoid re-looking up in the feature-set!

Reply:
since needed features come and go, there will be a time `FeeFeatures` becomes an empty struct. I would think to keep it empty then (perhaps with a note or something to prevent innocent "cleanup"), so to keep `calculate_fee()` signature unchanged. wdyt?

Reply:
yeah I agree with keeping it with a note, so that the design idea is not lost

Reply:
[9b75b1c](https://github.com/anza-xyz/agave/pull/4472/commits/9b75b1cbf42e4f00e3a3bc570b2783449e7254a2)